sound/soc/sof/control.c

```diff
@@ -416,6 +416,14 @@ int snd_sof_bytes_ext_put(struct snd_kcontrol *kcontrol,
 	if (copy_from_user(&header, tlvd, sizeof(const struct snd_ctl_tlv)))
 		return -EFAULT;
 
+	/* Check that length does not exceed the maximum defined for SOF */
+	if (header.length > SND_SOF_CTL_TLV_LENGTH_MAX) {
+		dev_err_ratelimited(sdev->dev,
+				    "error: Bytes size %d exceeds max %d.\n",
+				    header.length, SND_SOF_CTL_TLV_LENGTH_MAX);
```

Comment:
header.length is unsigned, to, I think the comparison will be unsigned too, so I'd use "%u" for printing header.length as well. Same below

Reply:
ok